test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -654,8 +654,23 @@ type_spec name alter = Test.group name <|
         alter [1, 1.0, 2, 2.0] . distinct . should_equal [1, 2]
         alter [] . distinct . should_equal []
 
+    Test.specify "should be able to handle distinct on different primitive values" <|
+        alter [1, "a"] . distinct . should_equal [1, "a"]
+        alter ["a", 1] . distinct . should_equal ["a", 1]
+        alter [Nothing, Nothing] . distinct . should_equal [Nothing]
+        alter [Number.nan, Number.nan] . distinct . to_text . should_equal "[NaN]"
+        alter [Nothing, Number.nan, Nothing, Number.nan] . distinct . to_text . should_equal "[Nothing, NaN]"
+        alter [1, Nothing] . distinct . should_equal [1, Nothing]
+        alter [Nothing, 1, Nothing] . distinct . should_equal [Nothing, 1]
+        alter [1, Number.nan] . distinct . to_text . should_equal "[1, NaN]"
+        alter [Number.nan, 1, Number.nan] . distinct . to_text . should_equal "[NaN, 1]"
+        alter [1, Number.nan, 1] . distinct . to_text . should_equal "[1, NaN]"
```

Comment:
How does `distinct` work with other non-reflexive values, like `My_Nan`? Can we get a test for it too?

Reply:
Added in 6bae7283fe1ce8276de282dcee37ef45abaa7d89